cli/resolver.rs

```diff
@@ -541,8 +545,23 @@ impl Resolver for CliGraphResolver {
       });
     let result = match result {
       Ok(resolution) => match resolution {
-        MappedResolution::Normal(specifier)
-        | MappedResolution::ImportMap(specifier) => {
+        MappedResolution::Normal {
+          specifier,
+          maybe_diagnostic,
+        }
+        | MappedResolution::ImportMap {
+          specifier,
+          maybe_diagnostic,
+        } => {
+          if let Some(diagnostic) = maybe_diagnostic {
+            match &*diagnostic {
+              MappedResolutionDiagnostic::ConstraintNotMatchedLocalVersion { reference, .. } => {
+                if self.warned_pkgs.insert(reference.req().clone()) {
+                  log::warn!("{} {}\n    at {}", colors::yellow("Warning"), diagnostic, referrer_range);
+                }
```

Comment:
Does the message say that this patch was not used due to not matching constraints?